src/components/DocLink.tsx

```diff
@@ -0,0 +1,27 @@
+import { useDocs } from "context/useDocs";
+import type { FC, ReactNode } from "react";
+
+interface Props {
+  docPath: string;
+  content: ReactNode;
+  className?: string;
+  title?: string;
+}
```

Comment:
```suggestion
interface Props {
  docPath: string;
  children: ReactNode;
  hasIcon?: boolean;
  className?: string;
  title?: string;
}
```
I think we should use the more native `children` to pass the content of the link.
Also, I think we can add a `hasIcon` prop with a default of `false` to also cover the scope of `ExternalDocLink` and remove that 2nd component. Then instead of `ExternalDocLink` we can do `<DocLink withIcon docPath={...`